basic_pacman/pacman_env.py

```diff
@@ -52,68 +169,57 @@ def __move(self):
         if self.orientation == 3:
             self.pos[1] -= 1
 
-        return self.pos
+        self.step_counter += 1
 
-    def step(self, action):
+    def _set_action(self, action: int | None) -> None:
         """
-        This function realizes the change in the game state according to the chosen action.
-        :param action: chosen action (integer 0/1/2/3 corresponding to the directions in order up/right/down/left)
-        :return: (state, reward, done, info)
+        This function coordinates the modification of the 'orientation' variable based on user and timer (To be
+        implemented...) input.
+        :param action: chosen action 0/1/2/3 corresponding to the directions in order up/right/down/left)
+        :return: None
         """
-        if 0 <= action < 4:
-            self.orientation = action
-        else:
-            raise ValueError
-
-        self.pos = self.__move()
-        self.step_cnt += 1
-
-        reward = 0
-
-        done = self._check_done()
+        self.orientation = action
+        # Timer response to be implemented later...
 
-        self._calculate_score()
-        info = "Points acquired: " + str(self.score)
-
-        self.state = self.update_map()
-
-        return self.state, reward, done, info
+    def _check_action_validity(self, action: int) -> None:
+        """
+        This function verifies the user input's validity.
+        :param action: chosen action 0/1/2/3 corresponding to the directions in order up/right/down/left)
+        :return: None
+        """
+        if min(self.action_space) <= action <= max(self.action_space):
+            return
+        else:
+            raise ValueError("Please enter a number in the [0; 4) interval!")
 
-    def _calculate_score(self):
+    def _calculate_score(self) -> None:
         """
         This function calculates the scores which may have been acquired during the last step, and if so, removes the
         coin from the map.
+        Score calculation:
+            +10 / coin collected
         :return: None
         """
-        for coin_pos in self.coins:
+        for coin_pos in self.coins_pos:
             if self.pos == coin_pos:
                 self.score += 10
```

Comment:
Maybe use a constant value for score calculation like you did it before (for example in case of NUM_GHOSTS)

```suggestion
                self.score += SCORE_FOR_COIN
```

Reply:
First of all, thank you @NyesteAmbrusPal for the feedback and the suggestion!
As it seemed quite reasonable not to use magic numbers, I added a constant to the code in a new commit that resolves this problem.